modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/spring/ObjectFactoryClassRepository.java

```diff
@@ -49,31 +66,67 @@
    * {@code Class} instance should not be reused for another {@link ComponentBuildingDefinition}.
    *
-   * @param componentBuildingDefinition          the definition on how to build the component
-   * @param objectFactoryType                    the {@link ObjectFactory} of the component
-   * @param createdObjectType                    the type of object created by the {@code ObjectFactory}
-   * @param isLazyInitFunction                   function that defines if the object created by the component can be created
-   *                                             lazily
-   * @param instancePostCreationFunctionOptional function to do custom processing of the created instance by the
-   *                                             {@code ObjectFactory}. When there's no need for post processing this value must
-   *                                             be {@link Optional#empty()}
+   * @param objectFactoryType the {@link ObjectFactory} of the component
    * @return the {@code FactoryBean} class to be used by spring for the provided configuration.
    */
-  public Class<ObjectFactory> getObjectFactoryClass(ComponentBuildingDefinition componentBuildingDefinition,
-                                                    Class objectFactoryType,
-                                                    Class createdObjectType,
-                                                    Supplier<Boolean> isLazyInitFunction) {
-    return getObjectFactoryDynamicClass(componentBuildingDefinition, objectFactoryType, createdObjectType, isLazyInitFunction);
+  public Class<ObjectFactory> getObjectFactoryClass(Class objectFactoryType) {
+    synchronized (this.getClass().getClassLoader()) {
+      String name = objectFactoryType.getName() + "_ByteBuddy";
+      ClassLoader classLoader = getClass().getClassLoader();
+      if (SmartFactoryBean.class.getClassLoader() != objectFactoryType.getClassLoader()) {
+        classLoader = COMPOSITE_CL_CACHE.get(objectFactoryType.getClassLoader());
+      }
+      try {
+        return (Class<ObjectFactory>) forName(name, true, classLoader);
+      } catch (ClassNotFoundException e) {
+        // class doesn't exist, generate
+      }
+      return createObjectFactoryDynamicClass(objectFactoryType, name, classLoader);
+    }
   }
 
-  private Class<ObjectFactory> getObjectFactoryDynamicClass(final ComponentBuildingDefinition componentBuildingDefinition,
-                                                            final Class objectFactoryType,
-                                                            final Class createdObjectType,
-                                                            final Supplier<Boolean> isLazyInitFunction) {
+  private Class<ObjectFactory> createObjectFactoryDynamicClass(Class objectFactoryType, String name, ClassLoader classLoader) {
+    final IsEagerInitGetterInterceptor interceptor = new IsEagerInitGetterInterceptor();
```

Comment:
how costly is it to instantiate a new IsEagerInitGetterInterceptor every time? Can the same instance be reused?